enzyme/Enzyme/MLIR/Implementations/MemRefAutoDiffOpInterfaceImpl.cpp

```diff
@@ -62,6 +65,75 @@ struct StoreOpInterface
   }
 };
 
+struct LoadOpInterfaceReverse : public AutoDiffOpInterfaceReverse::ExternalModel<LoadOpInterfaceReverse, memref::LoadOp> {
+  LogicalResult createReverseModeAdjoint(Operation *op, OpBuilder &builder, MGradientUtilsReverse *gutils) const {
+    auto loadOp = cast<memref::LoadOp>(op);
+    Value memref = loadOp.getMemref();
+    ValueRange vr = loadOp.getIndices();
+    
+    if (auto iface = loadOp.getType().cast<AutoDiffTypeInterface>()) {      
+      if(gutils->hasInvertPointer(loadOp) && gutils->hasInvertPointer(memref)){
+        OpBuilder cacheBuilder(gutils->getNewFromOriginal(op));
+        
+        Value gradient = gutils->invertPointerM(loadOp, builder);
+        Value memrefGradient = gutils->invertPointerM(memref, builder);
+
+        SmallVector<Value> retrievedArguments;
+        for (Value v : vr){
+          Value cache = gutils->cacheForReverse(gutils->getNewFromOriginal(v), cacheBuilder);
+          Value retrievedValue = gutils->popCache(cache, builder);
+          retrievedArguments.push_back(retrievedValue);
+        }
+
+        Value loadedGradient = builder.create<memref::LoadOp>(loadOp.getLoc(), memrefGradient, ArrayRef<Value>(retrievedArguments));
+        Value addedGradient = iface.createAddOp(builder, loadOp.getLoc(), loadedGradient, gradient);
+        builder.create<memref::StoreOp>(loadOp.getLoc(), addedGradient, memrefGradient, ArrayRef<Value>(retrievedArguments));
+      }   
+    }
+    return success();
+  }
+
+  void clearGradient(Operation *op, SmallVector<OpBuilder *>builders, MGradientUtilsReverse *gutils) const {
+  }
+};
+
+struct StoreOpInterfaceReverse : public AutoDiffOpInterfaceReverse::ExternalModel<StoreOpInterfaceReverse, memref::StoreOp> {
+  LogicalResult createReverseModeAdjoint(Operation *op, OpBuilder &builder, MGradientUtilsReverse *gutils) const {
+    auto storeOp = cast<memref::StoreOp>(op);
+    Value val = storeOp.getValue();
+    Value memref = storeOp.getMemref();
+    ValueRange vr = storeOp.getIndices();
```

Comment:
Ultra-nit: prefer more descriptive names than `vr` for long-lived variables. `Indices` is just 5 more characters to type.